src/components/ProgressForm/ProgressForm.tsx

```diff
@@ -13,6 +13,10 @@ import { useSaveProgressMutation } from '@/app/services/progressesApi';
 import { useRouter } from 'next/router';
 import Spinner from '../reusables/Spinner';
 
+interface ExtendedFormProps extends formProps {
+    onUpdateSuccess?: () => void;
+}
+
```

Comment:
Is formProps type getting used somewhere else?
Can't we add onUpdateSuccess inside formProps?
Also move the type to type file if we are maintaining it separately.

Reply:
I prefer keeping ExtendedFormProps separate because:

1. The base `formProps` type represents core form functionality shared across different contexts 
2. `onUpdateSuccess` is specific to the modal update flow and may not be relevant for all form usage scenarios
3. This separation maintains better modularity - we're just extending the base type without affecting other components using `formProps`
4. While we could add it to the base type, making it optional there would mean every consumer needs to handle that possibility, even if it's not relevant to their use case